github/messages.go

```diff
@@ -117,3 +120,41 @@ func validateSignature(signature string, payload, secretKey []byte) error {
 	}
 	return nil
 }
+
+// getWebhookEvent parses the payload and request, and build the Event.
+func getWebhookEvent(r *http.Request, payload []byte) (interface{}, error) {
+	eventType := r.Header.Get(eventTypeHeader)
+	if eventType == "push" {
+		pushEvent := WebHookPayload{}
+		if err := json.Unmarshal(payload, &pushEvent); err != nil {
+			return nil, err
+		}
+		return pushEvent, nil
+	}
+
+	event := Event{
+		Type:       &eventType,
+		RawPayload: (*json.RawMessage)(&payload),
+	}
+	return event.Payload(), nil
+}
+
+// ParseWebhook parses the request, validates the signature, and returns
+// an Event or a WebHookPayload.
+// secretKey is the GitHub Webhook secret message.
+//
+// Example usage:
+//
+//     func (s *GitHubEventMonitor) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+//       event, err := github.ParseWebHook(r, s.webhookSecretKey)
+//       if err != nil { ... }
+//       // Process event ...
+//     }
+//
+func ParseWebHook(r *http.Request, secretKey []byte) (interface{}, error) {
```

Comment:
If I look at the implementation of what `ParseWebHook` does, then look at what `getWebhookEvent` does, I'll see that it returns `Event.Payload`. Then if I refer to documentation for [`Event.Payload`](https://godoc.org/github.com/google/go-github/github#Event.Payload), it begins to make sense:

> Payload returns the parsed event payload. For recognized event types, a value of the corresponding struct type will be returned.

The docs of `ParseWebHook` just say "returns an Event or a WebHookPayload" which isn't as clear.